feat: add en anchors

wxsms opened the pull request:
## Description of Problem

close #1998 

added a script to generate anchors. It can also re-run for updates.

wxsms commented:
Looks like that `reactivity-fundamentals.md` has 2 headings with same name, which will generate same slug id and cause vitepress complain. Any idea?

edimitchel commented:
> Looks like that `reactivity-fundamentals.md` has 2 headings with same name, which will generate same slug id and cause vitepress complain. Any idea?

Seems to be a duplicate title for composition api / option api..

Jinjiang commented:
I suggest we just follow the current rule to add a incremental number after the duplicated ids. That's also what you can get on the current doc site.

![image](https://user-images.githubusercontent.com/206848/199775629-a30be0f7-18f1-4f3f-b397-2ae9f622571b.png)

For example, the second `#reactive-proxy-vs-original` should become `#reactive-proxy-vs-original-1` automatically.

Thanks.

skirtles-code commented:
I'm wondering whether we need to include the script in the PR? The script is useful for the initial bulk adding of headings, but do we really need to use it beyond that?

Perhaps we could just manually correct the handful of duplicate headers rather than trying to get the script to do it automatically?

edimitchel commented:
> I'm wondering whether we need to include the script in the PR? The script is useful for the initial bulk adding of headings, but do we really need to use it beyond that?
> 
> Perhaps we could just manually correct the handful of duplicate headers rather than trying to get the script to do it automatically?

It could be put into a gist instead of directly in the source of docs yes!

Jinjiang commented:
> > I'm wondering whether we need to include the script in the PR? The script is useful for the initial bulk adding of headings, but do we really need to use it beyond that?
> > Perhaps we could just manually correct the handful of duplicate headers rather than trying to get the script to do it automatically?
> 
> It could be put into a gist instead of directly in the source of docs yes!

Sounds a good idea.

wxsms commented:
I just removed the script related things from my branch. And fixed the duplicated anchor as @Jinjiang suggested.

the script posted here, and you may use it at will.

<details>

```js
/**
 * slugify function taken from vitepress
 */
// string.js slugify drops non ascii chars so we have to
// use a custom implementation here
const { remove: removeDiacritics } = require('diacritics')
// eslint-disable-next-line no-control-regex
const rControl = /[\u0000-\u001f]/g
const rSpecial = /[\s~`!@#$%^&*()\-_+=[\]{}|\\;:"'<>,.?/]+/g

const slugify = (str) => {
  return (
    removeDiacritics(str)
      // Remove control characters
      .replace(rControl, '')
      // Replace special characters
      .replace(rSpecial, '-')
      // Remove continuous separators
      .replace(/\-{2,}/g, '-')
      // Remove prefixing and trailing separators
      .replace(/^\-+|\-+$/g, '')
      // ensure it doesn't start with a number (#121)
      .replace(/^(\d)/, '_$1')
      // lowercase
      .toLowerCase()
  )
}

let glob = require('glob');
let path = require('path');
let fs = require('fs');

glob('../../src/**/*.md', { cwd: __dirname }, function (err, files) {
  for (let f of files) {
    let filePath = path.join(__dirname, f);
    let content = fs.readFileSync(filePath, { encoding: 'utf-8' });
    let headings = content.match(/^#+ .+$/gm);
    for (let h of headings ?? []) {
      let _h = h.replace(/{#.+$/, '').trim();
      content = content.replace(h, `${_h} {#${slugify(_h)}}`);
    }
    fs.writeFileSync(filePath, content, { encoding: 'utf-8' });
  }
});
```

</details>

Jinjiang commented on src/api/options-lifecycle.md:
```diff
@@ -209,7 +209,7 @@ Called when an error propagating from a descendant component has been captured.
   - An `errorCaptured` hook can return `false` to prevent the error from propagating further. This is essentially saying "this error has been handled and should be ignored." It will prevent any additional `errorCaptured` hooks or `app.config.errorHandler` from being invoked for this error.
 
-## renderTracked <sup class="vt-badge dev-only" />
+## renderTracked <sup class="vt-badge dev-only" /> {#rendertracked-sup-class-vt-badge-dev-only}
```
The original hash names like this one on the website (`#rendertracked`) actually doesn't include the `-sup-*` things.

https://vuejs.org/api/options-lifecycle.html#rendertracked

I think we'd better still follow this rule to avoid breaking changes.

Thanks.

wxsms replied:
fixed

skirtles-code commented on src/api/composition-api-dependency-injection.md:
```diff
@@ -1,6 +1,6 @@
-# Composition API: <br>Dependency Injection
+# Composition API: <br>Dependency Injection {#composition-api-br-dependency-injection}
```
I believe this has changed relative to the live docs, due to the `<br>`.

```suggestion
# Composition API: <br>Dependency Injection {#composition-api-dependency-injection}
```

hymair commented:
Seems like this introduced a regression:

<img width="494" alt="image" src="https://user-images.githubusercontent.com/21025134/200158956-3b744c95-f988-4bce-8f64-2e8aa9713133.png">

cc @wxsms @Jinjiang @skirtles-code @edimitchel

Jinjiang commented:
@hymair thanks for figuring it out. I created a PR to fix this. 🙏
